### DISCUSSION TITLE
Backlog via URL not working

### DISCUSSION THREAD

**Discussion by c-guenther** [arendst/Tasmota#18947] [Q&A] [answered]
### PROBLEM DESCRIPTION
Using a SONOFF CH4PROR3 - when trying to switch several Outputs via URL, it seems that only the first item after Backlog is being handeled.
URL is: http://192.168.2.90/cm?cmnd=Backlog%20Power1%20on;Power2%20on;Power3%20on;Power4%20on
When this command is given via Konsole all works as expected.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [X] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [X] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [X] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [X] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [X] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [X] Device used (e.g., Sonoff Basic): Sonoff CH4PROR3
- [X] Tasmota binary firmware version number used: 12.5.0
  - [X] Pre-compiled
  - [ ] Self-compiled OTA
- [X] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:
14:24:11.169 RSL: RESULT = {"NAME":"Sonoff 4CHPROR3","GPIO":[32,1,1,1,226,225,33,34,224,320,35,227,0,0],"FLAG":0,"BASE":23}
14:24:11.378 RSL: RESULT = {"Module":{"0":"Sonoff 4CHPROR3"}}
14:24:11.584 RSL: RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"226":"Relay3"},"GPIO5":{"225":"Relay2"},"GPIO9":{"33":"Button2"},"GPIO10":{"34":"Button3"},"GPIO12":{"224":"Relay1"},"GPIO13":{"320":"Led_i1"},"GPIO14":{"35":"Button4"},"GPIO15":{"227":"Relay4"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"}}


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [x] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
14:25:21.772 RSL: STATUS = {"Status":{"Module":0,"DeviceName":"Tasmota","FriendlyName":["Tasmota","Tasmota2","Tasmota3","Tasmota4"],"Topic":"tasmota_CB9ED1","ButtonTopic":"0","Power":1,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
14:25:21.807 RSL: STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-DE.bin.gz","RestartReason":"Power On","Uptime":"0T01:28:15","StartupUTC":"2023-06-23T11:57:06","Sleep":50,"CfgHolder":4617,"BootCount":5,"BCResetTime":"2023-06-23T12:33:16","SaveCount":71,"SaveAddress":"F5000"}}
14:25:21.840 RSL: STATUS2 = {"StatusFWR":{"Version":"12.5.0(tasmota)","BuildDateTime":"2023.04.17 08:08:43","Boot":7,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"395/699"}}
14:25:21.859 RSL: STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["WLAN XXXX",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A192800000000","00000080","00006000","00004000","00000000"]}}
14:25:21.893 RSL: STATUS4 = {"StatusMEM":{"ProgramSize":633,"Free":368,"Heap":25,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"14325E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000407","8F9AC787","04368001","000000CF","010013C0","C000F981","00004004","00001000","54000020","00000080"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,16,18,19,20,21,22,24,26,27,29,30,35,37,45,62","Sensors":"1,2,3,4,5,6","I2CDriver":"7"}}
14:25:21.925 RSL: STATUS5 = {"StatusNET":{"Hostname":"tasmota-CB9ED1-7889","IPAddress":"192.168.2.90","Gateway":"192.168.2.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.110","DNSServer2":"0.0.0.0","Mac":"4C:EB:D6:CB:9E:D1","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
14:25:21.948 RSL: STATUS6 = {"StatusMQT":{"MqttHost":"","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_CB9ED1","MqttUser":"DVES_USER","MqttCount":0,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
14:25:21.970 RSL: STATUS7 = {"StatusTIM":{"UTC":"2023-06-23T13:25:21","Local":"2023-06-23T14:25:21","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"04:47","Sunset":"20:57"}}
14:25:21.988 RSL: STATUS10 = {"StatusSNS":{"Time":"2023-06-23T14:25:21"}}
14:25:21.996 RSL: STATUS11 = {"StatusSTS":{"Time":"2023-06-23T14:25:21","Uptime":"0T01:28:15","UptimeSec":5295,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER1":"ON","POWER2":"OFF","POWER3":"OFF","POWER4":"OFF","Wifi":{"AP":1,"SSId":"WLAN XXX","BSSId":"0A:A0:57:47:9E:7E","Channel":11,"Mode":"11n","RSSI":88,"Signal":-56,"LinkCount":1,"Downtime":"0T00:00:20"}}}

```
- [x] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
14:27:03.537 HTP: Befehl
14:27:03.542 SRC: WebCommand from 192.168.1.111
14:27:03.545 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 9, Pld -99, Data 'Power1 on'
14:27:03.672 SRC: Backlog
14:27:03.675 CMD: Grp 0, Cmd 'POWER', Idx 1, Len 2, Pld 1, Data 'on'
14:27:03.679 RSL: RESULT = {"POWER1":"ON"}
14:27:03.681 RSL: POWER1 = ON

```

### TO REPRODUCE
Post URL http://192.168.2.90/cm?cmnd=Backlog%20Power1%20on;Power2%20on;Power3%20on;Power4%20on -> Only Output 1 will switch on.
Send command via Web-Console: All Outputs are switched on.

### EXPECTED BEHAVIOUR
With the URL given above, all outputs should come on.

### SCREENSHOTS
n/a

### ADDITIONAL CONTEXT
Have another CH4PRO working, Firmware 6.6.0 - there the problem is not existant.

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by sfromis** [comment, marked as answer]
You did not tell how you invoked that URL, but most likely you did it from a tool using `;` as some sort of delimiter, and that's why the docs recommend you to URL-encode all special chars, like using `%3b` for `;`, in the same way as you used `%20` for space.

As the log shows, Tasmota did clearly not receive anything after your first semicolon.

**Comment by c-guenther** [comment]
Gentlemen, f**k, please excuse my tomatoes on the eyes. shure, the not-url-encoded delimiter, what else. was distracted by the fact, that older fws seems to parse this without encoding - but, thank you for the hint!

**Reply by sfromis** [reply]
It was not about what older Tasmota vs newer Tasmota firmware is able to parse, but **only** about the tools you use when firing off the command including the URL. Tasmota does not care whether you URL-encode or not, as Tasmota does not need any URL-encoding, but without doing that, your URL may not "survive the travel" from your keyboard to being received by Tasmota.
